src/test/util/net.h

```diff
@@ -206,4 +209,155 @@ class StaticContentsSock : public ZeroSock
     mutable size_t m_consumed{0};
 };
 
+/**
+ * A mocked Sock alternative that allows providing the data to be returned by Recv()
+ * and inspecting the data that has been supplied to Send().
+ */
+class DynSock : public ZeroSock
+{
+public:
+    /**
+     * Unidirectional bytes or CNetMessage queue (FIFO).
+     */
+    class Pipe
+    {
+    public:
+        /**
+         * Get bytes and remove them from the pipe.
+         * @param[in] buf Destination to write bytes to.
+         * @param[in] len Write up to this number of bytes.
+         * @param[in] flags Same as the flags of `recv(2)`. Just `MSG_PEEK` is honored.
```

Comment:
@vasild these first two params are in/out right?

Reply:
Did you mean to comment on https://github.com/bitcoin/bitcoin/pull/30988? This branch is stale (though not sure if this method changed since).

Reply:
#50 uses SockMan and I try to keep it freshly rebased.

Reply:
ok ill start tracking there, just had a few quesitons for @vasild as I'm consuming DynSock in [my branch now too](https://github.com/pinheadmz/bitcoin/blob/d8158d80ce88f8584901bd9fa26f03aca76ffb80/src/test/httpserver_tests.cpp#L289)